File: codegen/swagger-ehr/dart/lib/model/reference_range.dart
part of swagger.api;

class ReferenceRange {
  
  DvText meaning = null;

  DvInterval range = null;

  ReferenceRange();

  @override
  String toString() {
    return 'ReferenceRange[meaning=$meaning, range=$range, ]';
  }

  ReferenceRange.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    meaning = new DvText.fromJson(json['meaning']);
    range = new DvInterval.fromJson(json['range']);
  }

  Map<String, dynamic> toJson() {
    return {
      'meaning': meaning,
      'range': range
     };
  }

  static List<ReferenceRange> listFromJson(List<dynamic> json) {
    return json == null ? new List<ReferenceRange>() : json.map((value) => new ReferenceRange.fromJson(value)).toList();
  }

  static Map<String, ReferenceRange> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, ReferenceRange>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new ReferenceRange.fromJson(value));
    }
    return map;
  }
}

File: codegen/oas-ehr/spring/src/main/java/org/openapitools/model/ReferenceRange.java
package org.openapitools.model;

import java.net.URI;
import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import org.openapitools.model.DvInterval;
import org.openapitools.model.DvText;
import org.openapitools.jackson.nullable.JsonNullable;
import java.time.OffsetDateTime;
import javax.validation.Valid;
import javax.validation.constraints.*;
import io.swagger.v3.oas.annotations.media.Schema;


import java.util.*;
import javax.annotation.Generated;

/**
 * ReferenceRange
 */

@Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
public class ReferenceRange {

  private DvText meaning;

  private DvInterval range;

  public ReferenceRange() {
    super();
  }

  /**
   * Constructor with only required parameters
   */
  public ReferenceRange(DvText meaning, DvInterval range) {
    this.meaning = meaning;
    this.range = range;
  }

  public ReferenceRange meaning(DvText meaning) {
    this.meaning = meaning;
    return this;
  }

  /**
   * Get meaning
   * @return meaning
  */
  @NotNull @Valid 
  @Schema(name = "meaning", requiredMode = Schema.RequiredMode.REQUIRED)
  @JsonProperty("meaning")
  public DvText getMeaning() {
    return meaning;
  }

  public void setMeaning(DvText meaning) {
    this.meaning = meaning;
  }

  public ReferenceRange range(DvInterval range) {
    this.range = range;
    return this;
  }

  /**
   * Get range
   * @return range
  */
  @NotNull @Valid 
  @Schema(name = "range", requiredMode = Schema.RequiredMode.REQUIRED)
  @JsonProperty("range")
  public DvInterval getRange() {
    return range;
  }

  public void setRange(DvInterval range) {
    this.range = range;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ReferenceRange referenceRange = (ReferenceRange) o;
    return Objects.equals(this.meaning, referenceRange.meaning) &&
        Objects.equals(this.range, referenceRange.range);
  }

  @Override
  public int hashCode() {
    return Objects.hash(meaning, range);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class ReferenceRange {\n");
    sb.append("    meaning: ").append(toIndentedString(meaning)).append("\n");
    sb.append("    range: ").append(toIndentedString(range)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

